File: core/feature47_test.go
package main

import "testing"

var (
	feature47Names      = []string{"ZMKS", "ZMGS", "KR", "GR", "RA", "L1", "L2", "M1", "M2"}
	feature47Accuracies = []string{"%.1f", "%.1f", "%.0f", "%.0f", "%.0f", "%.0f", "%.0f", "%.0f", "%.0f"}
	feature47Types      = []bool{true, true, false, false, false, false, false, false, false}
)

func Test01feature47(t *testing.T) {
	arguments := map[string]interface{}{"K": 200, "G": 300, "L": 200, "V": 200, "T": 6}
	expected := map[string]interface{}{"ZMKS": 104.7, "ZMGS": 157.1, "KR": []interface{}{500, 458, 361, 300}, "GR": []interface{}{750, 687, 541, 450}, "RA": []interface{}{600, 611}, "L1": []interface{}{250, 229, 180, 150}, "L2": []interface{}{218, 87, 132}, "M1": []interface{}{150, 76}, "M2": []interface{}{93}}
	genericTester(t, feature47(arguments), expected, feature47Names, feature47Accuracies, feature47Types)
}

File: core/feature47_test.js
var feature47Names = ["ZMKS", "ZMGS", "KR", "GR", "RA", "L1", "L2", "M1", "M2"];
var feature47Accuracies = [1, 1, 0, 0, 0, 0, 0, 0, 0];
var feature47Types = [true, true, false, false, false, false, false, false, false];

function test01feature47() {
  var arguments = {"K": 200, "G": 300, "L": 200, "V": 200, "T": 6};
  var expected = {"ZMKS": 104.7, "ZMGS": 157.1, "KR": [500, 458, 361, 300], "GR": [750, 687, 541, 450], "RA": [600, 611], "L1": [250, 229, 180, 150], "L2": [218, 87, 132], "M1": [150, 76], "M2": [93]};
  genericTester(feature47_core(arguments), expected, feature47Names, feature47Accuracies, feature47Types);
}

test01feature47();
